src/BasisTheory.net/Common/BaseClient.cs

```diff
@@ -170,6 +175,14 @@ private void SetRequestHeaders(HttpRequestMessage message, RequestOptions reques
 
             if (!string.IsNullOrEmpty(requestOptions?.CorrelationId))
                 message.Headers.Add("BT-TRACE-ID", requestOptions.CorrelationId);
+            
+            var btClientUserAgentString = UserAgentUtility.BuildBtClientUserAgentString(AppInfo);
+            if (!string.IsNullOrEmpty(btClientUserAgentString))
+                message.Headers.Add("BT-CLIENT-USER-AGENT", btClientUserAgentString);
+
+            var userAgentString = UserAgentUtility.BuildUserAgentString(AppInfo);
+            if (!string.IsNullOrEmpty((userAgentString)))
+                message.Headers.Add("User-Agent", userAgentString);
```

Comment:
These headers don't change on a per-request basis. Would it be better to set the DefaultRequestHeaders on the HttpClient that gets passed in or after we build the default client? That would save the overhead of computing the user agent strings per request.

Reply:
The only thing is that you can pass information about your application when creating the different clients (Application/Tokens/etc). Would have to think of a good way to do that with the other approach...